src/utils/hooks/useIsMountedRef.js

```diff
@@ -0,0 +1,18 @@
+import { useRef, useEffect } from 'react';
+
+// used mainly to prevent updates after unmount
+const useIsMountedRef = () => {
+  const isMountedRef = useRef(true);
+
+  useEffect(() => {
+    isMountedRef.current = true;
+
+    return () => {
+      isMountedRef.current = false;
+    };
+  }, []);
+
+  return isMountedRef;
```

Comment:
This will only handles a boolean value and nothing more inside the `current`. Perhaps you could return `isMountedRef.current` directly. That will give you the direct access to the boolean instead of doing that every time you want to use the hook value.

Reply:
If you do this, consider renaming the hook to `useIsMounted`.